source/deploy/postgres-migration.rst

```diff
@@ -128,39 +127,68 @@ Full-text indexes
 
 It's possible that some words in the ``Posts`` and ``FileInfo`` tables can exceed the `limits of the maximum token length <https://www.postgresql.org/docs/11/textsearch-limitations.html>`__ for full-text search indexing. In these cases, we are dropping the ``idx_posts_message_txt`` and ``idx_fileinfo_content_txt`` indexes from the PostgreSQL schema, and creating these indexes after the migration by running the following queries:
 
-Although these statements are included in the script, we recommend running these manually to prevent errors:
+To prevent errors during the migration, we have included following queries:
 
 .. code:: sql
 
    DROP INDEX IF EXISTS {{ .source_schema }}.idx_posts_message_txt;
    DROP INDEX IF EXISTS {{ .source_schema }}.idx_fileinfo_content_txt;
 
-The following queries are added to the script to re-create these indexes after migration finishes:
+To avoid performance regression on ``Posts`` and ``FileInfo`` table access, following queries should be executed once the migration finishes:
 
 .. code:: sql
 
    CREATE INDEX IF NOT EXISTS idx_posts_message_txt ON {{ .source_schema }}.posts USING gin(to_tsvector('english', message));
    CREATE INDEX IF NOT EXISTS idx_fileinfo_content_txt ON {{ .source_schema }}.fileinfo USING gin(to_tsvector('english', content));
 
 .. note::
-   If any of the entries in your  ``Posts`` and ``FileInfo`` tables exceed the limit, ``pgloader`` will fail with the ``ERROR:  string is too long for tsvector`` error while trying to create these indexes. You should manually remove these indexes.
+   If any of the entries in your  ``Posts`` and ``FileInfo`` tables exceed the limit mentioned above, index creation query will warn with the ``ERROR:  string is too long for tsvector`` log while trying to create these indexes.
```

Comment:
I think we should also mention the query to create the index based on truncated content.

Reply:
There are  two issues here though, to be more precise two kind of errors:
- words being longer than `2047` characters
- entire string being larger than  `1048575` bytes

So you can't easily truncate for words IIUC. You would just truncate entire string for a single occurrence in the entire text if you receive an error `word is too long to be indexed`.

cc @agnivade

Reply:
Yes correct. I think handling the first case is more tricky. I was suggesting to mention the query to handle the second case.

Reply:
I added a couple of sentenced for that, do you think that makes sense?

Reply:
I think customers will eventually ask for "what do I need to run to fix this?" So it might be better to provide a SQL query.